### PR TITLE
Implement topics needed by the UI for road weather use case

### PR DESCRIPTION
**Pull request by dan-du-car**
# PR Details
## Description
Implement topics needed by the UI by leveraging the existing UI layouts

## Motivation and Context
Road Weather use case

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: carma_wm_ctrl/src/WMBroadcaster.cpp [hunk 1]
```diff
@@ -245,6 +259,14 @@ void WMBroadcaster::addRegionAccessRule(std::shared_ptr<Geofence> gf_ptr, const
   gf_ptr->regulatory_element_ = regulatory_element;
 }
 
+void WMBroadcaster::addRegionMinimumGap(std::shared_ptr<Geofence> gf_ptr, double min_gap, const std::vector<lanelet::Lanelet>& affected_llts, const std::vector<lanelet::Area>& affected_areas) const
+{
+  auto regulatory_element = std::make_shared<lanelet::DigitalMinimumGap>(lanelet::DigitalMinimumGap::buildData(lanelet::utils::getId(), 
+                                        min_gap, affected_llts, affected_areas, { lanelet::Participants::VehicleCar }));
```
Comment (msmcconnell) [inline carma_wm_ctrl/src/WMBroadcaster.cpp:265]: The participant list you have here should be based on the TrafficControlParam.vclasses list. You can use the  participantsChecker() method to extract that from the control message. There is also a invertParticipants() method if that is needed.

File: carma_wm_ctrl/src/WMBroadcaster.cpp [hunk 2]
```diff
@@ -868,18 +890,35 @@ lanelet::BasicPoint2d curr_pos;
...
             outgoing_geof.is_on_active_geofence = true;
             for (auto regem: current_llt.regulatoryElements())
               {
                   if (regem->attribute(lanelet::AttributeName::Subtype).value().compare(lanelet::DigitalSpeedLimit::RuleName) == 0)
                   {
                     lanelet::DigitalSpeedLimitPtr speed =  std::dynamic_pointer_cast<lanelet::DigitalSpeedLimit>
                     (current_map_->regulatoryElementLayer.get(regem->id()));
-                   outgoing_geof.value = speed->speed_limit_.value();
+                    outgoing_geof.value = speed->speed_limit_.value();
+                    outgoing_geof.advisory_speed = speed->speed_limit_.value();
```
Comment (msmcconnell) [inline carma_wm_ctrl/src/WMBroadcaster.cpp:902]: What is the difference in this outgoing message between .value and .advisory_speed?

**Review by msmcconnell — CHANGES_REQUESTED** [review]
This is looking really good Dan. Nice Job! Just 2 comments

File: carma_wm_ctrl/src/WMBroadcaster.cpp (hunk 2 above)
Comment (dan-du-car) [reply carma_wm_ctrl/src/WMBroadcaster.cpp:902]: value is only used for Speed Limit event type. The Advisory speed is for the closed event type. The UI is taking data from different speed fields depends on the event types.

Comment (msmcconnell) [reply carma_wm_ctrl/src/WMBroadcaster.cpp:902]: ok sounds good

Comment (dan-du-car) [reply carma_wm_ctrl/src/WMBroadcaster.cpp:902]: https://github.com/usdot-fhwa-stol/carma-platform/pull/1197/commits/761f745a65683501b99f5387e17f2388de375e83

File: carma_wm_ctrl/src/WMBroadcaster.cpp (hunk 1 above)
Comment (dan-du-car) [reply carma_wm_ctrl/src/WMBroadcaster.cpp]: https://github.com/usdot-fhwa-stol/carma-platform/pull/1197/commits/761f745a65683501b99f5387e17f2388de375e83

File: carma_wm_ctrl/src/WMBroadcaster.cpp [hunk 3]
```diff
@@ -259,10 +259,10 @@ void WMBroadcaster::addRegionAccessRule(std::shared_ptr<Geofence> gf_ptr, const
   gf_ptr->regulatory_element_ = regulatory_element;
 }
 
-void WMBroadcaster::addRegionMinimumGap(std::shared_ptr<Geofence> gf_ptr, double min_gap, const std::vector<lanelet::Lanelet>& affected_llts, const std::vector<lanelet::Area>& affected_areas) const
+void WMBroadcaster::addRegionMinimumGap(std::shared_ptr<Geofence> gf_ptr,  const cav_msgs::TrafficControlMessageV01& msg_v01, double min_gap, const std::vector<lanelet::Lanelet>& affected_llts, const std::vector<lanelet::Area>& affected_areas) const
 {
   auto regulatory_element = std::make_shared<lanelet::DigitalMinimumGap>(lanelet::DigitalMinimumGap::buildData(lanelet::utils::getId(), 
-                                        min_gap, affected_llts, affected_areas, { lanelet::Participants::VehicleCar }));
+                                        min_gap, affected_llts, affected_areas, invertParticipants(participantsChecker(msg_v01))));
```
Comment (msmcconnell) [inline carma_wm_ctrl/src/WMBroadcaster.cpp:265]: Looking over the implementation, I don't think you need to invert the provided participants. The message specified which participants are restricted and your implementation of DigitalMinimumGap works the same way so I think a call to participantChecker() is sufficient.

**Review by msmcconnell — CHANGES_REQUESTED** [review]

File: carma_wm_ctrl/src/WMBroadcaster.cpp (hunk 3 above)
Comment (dan-du-car) [reply carma_wm_ctrl/src/WMBroadcaster.cpp]: https://github.com/usdot-fhwa-stol/carma-platform/pull/1197/commits/b61c21edc31aa0b79c0e2e72ad8ddcd2f182faa8

**Review by msmcconnell — APPROVED** [review]
Looks good to me. Will wait for CI to show build is valid before merging

**Comment by msmcconnell** [comment]
This PR resolves 
https://github.com/usdot-fhwa-stol/carma-platform/issues/1205

**Pull request merged**